src/Pages/TemplatesTable/components/TemplateFilters.tsx

```diff
@@ -0,0 +1,281 @@
+  // Needed to fix styling when "Add repositories" button is disabled
+  repositoryActions: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+});
+
+export type Filters = 'Search' | 'Version' | 'Architecture';
+
+const Filters = ({ isLoading, setFilterData, filterData }: Props) => {
+  const classes = useStyles();
+  const { rbac } = useAppContext();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const filters = ['Search', 'Version', 'Architecture'];
+  const [filterType, setFilterType] = useState<Filters>('Search');
+  const [versionNamesLabels, setVersionNamesLabels] = useState({});
+  const [archNamesLabels, setArchNamesLabels] = useState({});
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedVersion, setSelectedVersion] = useState<string>('');
+  const [selectedArch, setSelectedArch] = useState<string>('');
+
+  const { distribution_arches = [], distribution_versions = [] } =
+    queryClient.getQueryData<RepositoryParamsResponse>(REPOSITORY_PARAMS_KEY) || {};
+
+  const clearFilters = () => {
+    setFilterType('Search');
+    setSearchQuery('');
+    setSelectedVersion('');
+    setSelectedArch('');
+    setFilterData({ search: '', version: '', arch: '' });
+  };
+
+  useEffect(() => {
+    // If the filters get cleared at the top level, sense that and clear them here.
+    if (!filterData.arch && !filterData.version && !filterData.search) {
+      clearFilters();
+    }
+  }, [filterData.arch, filterData.version, filterData.search]);
+
+  const {
+    searchQuery: debouncedSearchQuery,
+    selectedVersion: debouncedSelectedVersion,
+    selectedArch: debouncedSelectedArch,
+  } = useDebounce({
+    searchQuery,
+    selectedVersion,
+    selectedArch,
+  });
+
+  const getLabels = (type: 'arch' | 'version', name: string): string => {
+    const namesLabels = type === 'arch' ? distribution_arches : distribution_versions;
+    const found = namesLabels.find((v) => v.name === name);
+    if (found) {
+      return found.label;
+    }
+    return name;
+  };
+
+  useEffect(() => {
+    setFilterData({
+      search: debouncedSearchQuery,
+      version: getLabels('version', debouncedSelectedVersion),
+      arch: getLabels('arch', debouncedSelectedArch),
+    });
+  }, [debouncedSearchQuery, debouncedSelectedVersion, debouncedSelectedArch]);
+
+  useEffect(() => {
+    if (
+      isEmpty(versionNamesLabels) &&
+      isEmpty(archNamesLabels) &&
+      distribution_arches.length !== 0 &&
+      distribution_versions.length !== 0
+    ) {
+      const arches = {};
+      const versions = {};
+      distribution_arches.forEach((arch) => (arches[arch.name] = arch.label));
+      distribution_versions.forEach((version) => (versions[version.name] = version.label));
+      setVersionNamesLabels(versions);
+      setArchNamesLabels(arches);
+    }
+  }, [distribution_arches, distribution_versions]);
+
+  const Filter = useMemo(() => {
+    switch (filterType) {
+      case 'Search':
```

Comment:
should this be 'Name' ? it's unclear from 'Search' what it's filtering on, but i think it's by name

Reply:
Hmmm I was unsure myself if it was on both Name/Description or just name.

Reply:
all of my template descriptions have the word `description`, searching for `d` doesn't find them so it seems it's by name. i see the same Filter by Search for repositories in the add / edit template modals too, i think maybe those should be Filter by Name/URL?

Reply:
Good catch thanks!